Framework/CurveFitting/src/Algorithms/ConvertToYSpace.cpp

```diff
@@ -61,24 +61,23 @@ DetectorParams ConvertToYSpace::getDetectorParameters(
     throw std::invalid_argument(
         "ConvertToYSpace - Workspace has no source/sample.");
   }
-  Geometry::IDetector_const_sptr det;
-  try {
-    det = ws->getDetector(index);
-  } catch (Kernel::Exception::NotFoundError &) {
+
+  const auto &spectrumInfo = ws->spectrumInfo();
```

Comment:
We retrieve several other parameters from ws though, I'm not sure this is really necessary.